src/main.js

```diff
@@ -71,20 +77,68 @@ function checkEventTarget(event) {
 
 function deleteIdea(target) {
   var id = parseInt(target.parentNode.parentNode.id);
+  // var idea = localStorage.key(id);
     for (var i = 0; i < savedIdeas.length; i++) {
       if (savedIdeas[i].id === id) {
+        savedIdeas[i].deleteFromStorage();
         savedIdeas.splice(i, 1);
       };
     };
-    showIdeas();
+    showIdeas(savedIdeas);
 };
 
 function favoriteIdea(target) {
   var id = parseInt(target.parentNode.parentNode.id);
     for (var i = 0; i < savedIdeas.length; i++) {
       if (savedIdeas[i].id === id) {
         savedIdeas[i].star = !savedIdeas[i].star;
+        var idea = JSON.parse(localStorage.getItem(id));
+        idea.star = !idea.star;
+        localStorage.setItem(id, JSON.stringify(idea));
       };
     };
-    showIdeas()
+    showIdeas(savedIdeas)
 };
+
+function pageLoad() {
+  if (localStorage) {
+    for (var i = 0; i < localStorage.length; i++){
+      var key = localStorage.key(i);
+      var idea = JSON.parse(localStorage.getItem(key));
+      var id = idea.id;
+      savedIdeas.push(new Idea(idea.title, idea.body, id, idea.star));
+    }
+    showIdeas(savedIdeas);
+  }
+}
+
+
+function showStarredIdeas() {
+  searchInput.value = '';
+  if (showStarredButton.innerText === "Show Starred Ideas"){
+    showStarredButton.innerText = "Show All Ideas"
+    filterStarredIdeas(savedIdeas);
+  }else{
+    showStarredButton.innerText = "Show Starred Ideas"
+    showIdeas(savedIdeas);
+  }
+}
+
+function filterStarredIdeas(array){
```

Comment:
Is this function doing the same thing as line 134? It seems to be achieving the same outcome.